src/main/java/mekanism/common/block/BlockBounding.java

```diff
@@ -76,20 +86,82 @@ public ItemStack getPickBlock(IBlockState state, RayTraceResult target, World wo
 			IBlockState state1 = world.getBlockState(tileEntity.mainPos);
 			return state1.getBlock().getPickBlock(state1, target, world, tileEntity.mainPos, player);
 		} catch(Exception e) {
+			e.printStackTrace();
 			return ItemStack.EMPTY;
 		}
 	}
 
+	/**
+	 * {@inheritDoc}
+	 * Delegate to main {@link Block#getDrops(NonNullList, IBlockAccess, BlockPos, IBlockState, int)}.
+	 */
 	@Override
-	public boolean removedByPlayer(IBlockState state, World world, BlockPos pos, EntityPlayer player, boolean willHarvest)
+	public void getDrops(NonNullList<ItemStack> drops, IBlockAccess world, BlockPos pos,IBlockState state, int fortune)
 	{
 		try {
 			TileEntityBoundingBlock tileEntity = (TileEntityBoundingBlock)world.getTileEntity(pos);
-			IBlockState state1 = world.getBlockState(tileEntity.mainPos);
-			return state1.getBlock().removedByPlayer(state1, world, tileEntity.mainPos, player, willHarvest);
+			BlockPos mainPos = tileEntity.mainPos;
+			IBlockState state1 = world.getBlockState(mainPos);
+			state1.getBlock().getDrops(drops, world, mainPos, state1, fortune);
 		} catch(Exception e) {
-			return false;
+			e.printStackTrace();
+		}
+	}
+
+	/**
+	 * {@inheritDoc}
+	 * Keep tile entity in world until after
+	 * {@link Block#getDrops(NonNullList, IBlockAccess, BlockPos, IBlockState, int)}.
+	 * Used together with {@link Block#harvestBlock(World, EntityPlayer, BlockPos, IBlockState, TileEntity, ItemStack)}.
+	 *
+	 * @author Forge
+	 * @see BlockFlowerPot#removedByPlayer(IBlockState, World, BlockPos, EntityPlayer, boolean)
+	 */
+	@Override
+	public boolean removedByPlayer(IBlockState state, World world, BlockPos pos, EntityPlayer player,
+	                               boolean willHarvest)
+	{
+		if (willHarvest) {
+			return true;
 		}
+
+		removeMainBlock(world, pos);
+		return super.removedByPlayer(state, world, pos, player, willHarvest);
+	}
+
+	private void removeMainBlock(World world, BlockPos pos) {
+		TileEntityBoundingBlock tileEntity = (TileEntityBoundingBlock)world.getTileEntity(pos);
+		world.setBlockToAir(tileEntity.mainPos);
```

Comment:
Setting the main block to air doesnt seem right to me...

Reply:
It works fine, because we drop the items in ```getDrops``` on behalf of the main block.

The main block shouldn't mind not being told about being removed, because all the ```removedByPlayer``` and ```harvestBlock``` hooks were just added.
All of the other bounding blocks are removed with block updates.

Reply:
> It works fine, because we drop the items in getDrops on behalf of the main block.

That is bad practice. The main block should be handling its own breakage.

Reply:
It's not the best but if the main block were to handle its own harvesting, it would drop itself as an itemstack into the world.
That means I can't pack up my base with RFTools builder anymore (or probably even Buildcraft quarry), which is the purpose of this pull request.
Another way is to add a method to the main block, to ask it to give us the items, but isn't that just ```getDrops```?

Reply:
Doing a ```getDrops``` and ```setBlockToAir``` seems to be standard, and will be done to us by other mods:
1. https://github.com/McJty/RFTools/blob/496527a1faa37ad16b60d2ea10751074c3d7d2e8/src/main/java/mcjty/rftools/blocks/builder/BuilderTileEntity.java#L1136-L1144
2. https://findusages.com/search/net.minecraft.block.Block/getDrops$4
3. https://findusages.com/search/net.minecraft.block.Block/getDrops$5